x-pack/packages/kbn-elastic-assistant/impl/assistant/settings/assistant_settings.test.tsx

```diff
@@ -0,0 +1,133 @@
+ */
+
+import { alertConvo, customConvo, welcomeConvo } from '../../mock/conversation';
+import { useAssistantContext } from '../../assistant_context';
+import { fireEvent, render } from '@testing-library/react';
+import {
+  AssistantSettings,
+  ANONYMIZATION_TAB,
+  CONVERSATIONS_TAB,
+  EVALUATION_TAB,
+  KNOWLEDGE_BASE_TAB,
+  QUICK_PROMPTS_TAB,
+  SYSTEM_PROMPTS_TAB,
+} from './assistant_settings';
+import React from 'react';
+import { OpenAiProviderType } from '@kbn/stack-connectors-plugin/common/openai/constants';
+import { MOCK_QUICK_PROMPTS } from '../../mock/quick_prompt';
+
+const mockConversations = {
+  [alertConvo.id]: alertConvo,
+  [welcomeConvo.id]: welcomeConvo,
+};
+const saveSettings = jest.fn();
+
+const mockValues = {
+  conversationSettings: mockConversations,
+  saveSettings,
+};
+
+const setSelectedSettingsTab = jest.fn();
+const mockContext = {
+  basePromptContexts: MOCK_QUICK_PROMPTS,
+  setSelectedSettingsTab,
+  http: {},
+  modelEvaluatorEnabled: true,
+  selectedSettingsTab: 'CONVERSATIONS_TAB',
+};
+const onClose = jest.fn();
+const onSave = jest.fn();
+const setSelectedConversationId = jest.fn();
+
+const testProps = {
+  defaultConnectorId: '123',
+  defaultProvider: OpenAiProviderType.OpenAi,
+  selectedConversation: welcomeConvo,
+  onClose,
+  onSave,
+  setSelectedConversationId,
+};
+jest.mock('../../assistant_context');
+
+jest.mock('./settings', () => {
+  return {
+    AnonymizationSettings: () => <span data-test-subj="ANONYMIZATION_TAB-tab" />,
+    ConversationSettings: () => <span data-test-subj={`CONVERSATION_TAB-tab`} />,
+    EvaluationSettings: () => <span data-test-subj="EVALUATION_TAB-tab" />,
+    KnowledgeBaseSettings: () => <span data-test-subj="KNOWLEDGE_BASE_TAB-tab" />,
+    QuickPromptSettings: () => <span data-test-subj="QUICK_PROMPTS_TAB-tab" />,
+    SystemPromptSettings: () => <span data-test-subj="SYSTEM_PROMPTS_TAB-tab" />,
+  };
+});
+
+jest.mock('./use_settings_updater/use_settings_updater', () => {
+  const original = jest.requireActual('./use_settings_updater/use_settings_updater');
+  return {
+    ...original,
+    useSettingsUpdater: jest.fn().mockImplementation(() => mockValues),
+  };
+});
+
+describe('AssistantSettings', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    (useAssistantContext as jest.Mock).mockImplementation(() => mockContext);
+  });
+
+  it('saves changes', () => {
+    const { getByTestId } = render(<AssistantSettings {...testProps} />);
+    // fireEvent.click(getByTestId(`${EVALUATION_TAB}-button`));
+    fireEvent.click(getByTestId('save-button'));
+    expect(onSave).toHaveBeenCalled();
+    expect(saveSettings).toHaveBeenCalled();
+  });
+
+  it('saves changes and updates selected conversation when selected conversation has been deleted', () => {
+    const { getByTestId } = render(
+      <AssistantSettings {...testProps} selectedConversation={customConvo} />
+    );
+    // fireEvent.click(getByTestId(`${EVALUATION_TAB}-button`));
```

Comment:
ditto above: This lingering around for a future `evaluation` test? Maybe not necessary since the eval tab doesn't contain any savable state?

```suggestion
```